tests/ipfwd/test_mtu.py

```diff
@@ -5,14 +5,111 @@
 from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory   # lgtm[py/unused-import]
 from tests.ptf_runner import ptf_runner
 from datetime import datetime
+from ipaddress import ip_address
+import json
+logger = logging.getLogger(__name__)
 
 pytestmark = [
-    pytest.mark.topology('t1')
+    pytest.mark.topology('t1', 't2')
 ]
 
+'''
+In case of multi-dut we need src_host_ip, src_router_ip, dst_host_ip, src_ptf_port_list, dst_ptf_port_list for the dut under test, 
+to take care of that made changes in the testcase 
+'''
+def get_lag_facts(dut, lag_facts, mg_facts, ignore_lags, test_facts, key='src'):
+    if not mg_facts['minigraph_portchannels']:
+        pytest.fail("minigraph_portchannels is not defined")
+
+    # minigraph facts
+    up_lag = None
+    for a_lag_name, a_lag_data in lag_facts['lags'].items():
+        if a_lag_data['po_intf_stat'] == 'Up' and a_lag_name not in ignore_lags:
+            # We found a portchannel that is up.
+            up_lag = a_lag_name
+            test_facts[key + '_port_ids'] = [mg_facts['minigraph_ptf_indices'][intf] for intf in a_lag_data['po_config']['ports']]
+            for intf in mg_facts['minigraph_portchannel_interfaces']:
+                if intf['attachto'] == up_lag:
+                    addr = ip_address(unicode(intf['addr']))
+                    if addr.version == 4:
+                        test_facts[key + '_router_ip'] = intf['addr']
+                        test_facts[key + '_host_ip'] = intf['peer_addr']
+                        break
+            logger.info("{} lag is {}".format(key, up_lag))
+            break
+
+    return up_lag
+
+
+def get_port_facts(dut, mg_facts, port_status, ignore_intfs, test_facts, key='src'):
+    if not mg_facts['minigraph_interfaces']:
+        pytest.fail("minigraph_interfaces is not defined.")
+
+    up_port = None
+    for a_intf_name, a_intf_data in port_status['int_status'].items():
+        if a_intf_data['oper_state'] == 'up' and a_intf_name not in ignore_intfs:
+            # Got a port that is up and not already used.
+            for intf in mg_facts['minigraph_interfaces']:
+                if intf['attachto'] == a_intf_name:
+                    up_port = a_intf_name
+                    addr = ip_address(unicode(intf['addr']))
+                    if addr.version == 4:
+                        test_facts[key + '_router_ip'] = intf['addr']
+                        test_facts[key + '_host_ip'] = intf['peer_addr']
+                        test_facts[key + '_port_ids'] = [mg_facts['minigraph_ptf_indices'][a_intf_name]]
+                        break
+            if up_port:
+                logger.info("{} port is {}".format(key, up_port))
+                break
+    return up_port
+
+@pytest.fixture(scope='function')
+def gather_facts(tbinfo, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
```

Comment:
It looks like there are code duplication in the updated two test scripts. Can you avoid duplication of functions and fixtures like:
* get_lag_facts
* get_port_facts
* gather_facts

Reply:
Moved to conftest.py as per the comment.

Reply:
could you please review the changes?